File: HW4/java/main/Main.java
package main;

import java.io.*;
import lexer.*;

public class Main {
	public static void main(String[] args) throws IOException
	{
		System.setIn(new FileInputStream("Test.txt"));
		Lexer lexer = new Lexer();
		try {
			while (true) {
				Token t = lexer.scan();
				System.out.println(t.toString());
			}
		} catch (IOException e) {
			System.out.println(e.getMessage());
		}
	}
}

File: HW4/cpp/Main.cpp
#include <iostream>

#include "Tag.h"
#include "Token.h"
#include "Word.h"
#include "Type.h"
#include "Num.h"
#include "Real.h"
#include "Lexer.h"

using namespace std;

int main(int argc, char** argv) throw(string)
{
	Lexer lexer = Lexer();
	try {
		while (true) {
			Token *t = lexer.scan();
			cout << t->toString() << endl;
			delete t;
		}
	} catch (string e) {
		cout << e << endl;
	}
	
	return 0;
}
